src/Sylius/Bundle/ApiBundle/Validator/Constraints/ShopUserResetPasswordTokenNotExpiredValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use Sylius\Component\User\Model\UserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class ShopUserResetPasswordTokenNotExpiredValidator extends ConstraintValidator
+{
+    /**
+     * @param UserRepositoryInterface<UserInterface> $userRepository
+     */
+    public function __construct(
+        private UserRepositoryInterface $userRepository,
+        private string $passwordResetTokenTtl,
+    ) {
+    }
+
+    /** @param string $value */
+    public function validate($value, Constraint $constraint): void
+    {
+        Assert::string($value);
+
+        /** @var ShopUserResetPasswordTokenNotExpired $constraint */
+        Assert::isInstanceOf($constraint, ShopUserResetPasswordTokenNotExpired::class);
+
+        /** @var UserInterface|null $user */
+        $user = $this->userRepository->findOneBy(['passwordResetToken' => $value]);
+
+        if (null === $user) {
+            return;
+        }
+
+        $lifetime = new \DateInterval($this->passwordResetTokenTtl);
```

Comment:
Can be done at constructor to validate TTL string validness early. Not sure about downsides.

P.S. ttl string is not being validated before as far as I see.
https://github.com/Sylius/Sylius/blob/cacced4b484e0da9c1f63edfbbdf3e19cdd1f758/src/Sylius/Bundle/UserBundle/DependencyInjection/Configuration.php#L67

Reply:
@diimpp can you propose a way to validate the dateinterval in the configuration file ?

Reply:
Sorry for late reply, yes, it's better to handle this in separate PR for UserBundle configuration, for example like this

```php
$resolver->setAllowedValues('ttl', function (string $value): bool {
    try {
       new \DateInterval($value);

       return true;
    } catch (\Exception $e) {
        return false;
    }
});
```